src/sentry/api/serializers/models/organization_member/expand/roles.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+from typing import Any, Iterable, Mapping, Sequence
+
+from sentry import roles
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models.role import RoleSerializer
+from sentry.api.serializers.models.user import DetailedUserSerializer
+from sentry.models import OrganizationMember, User
+from sentry.roles.manager import Role
+
+from .. import OrganizationMemberWithTeamsSerializer
+
+
+class OrganizationMemberWithRolesSerializer(OrganizationMemberWithTeamsSerializer):
+    def __init__(
+        self,
+        can_admin: bool,
+        allowed_roles: Iterable[Role],
+        expand: Sequence[str] | None = None,
+    ) -> None:
+        super().__init__(expand)
+        self.can_admin = can_admin
+        self.allowed_roles = allowed_roles
+
+    def serialize(  # type: ignore
+        self,
+        obj: OrganizationMember,
+        attrs: Mapping[str, Any],
+        user: User,
+        **kwargs: Any,
+    ) -> Mapping[str, Any]:
```

Comment:
could we type this with a `TypedDict` similar to the other om serializers, and update the serializers in the API annotations for the member_details endpoints? thanks!